### PR TITLE
Currency unit tests

### PR DESCRIPTION
**Pull request by hensha256**

### CODE REVIEW

**Comment by snreynolds** [comment]
It looks like Currency.sol doesn't have full coverage - is that a mistake of the tool? You can run `forge coverage --report debug` to inspect what branches in the code are untested.

**Comment by dianakocsis** [comment]
> forge coverage --report debug

I think it might be mistake of the tool. 
`Uncovered for src/types/Currency.sol:
- Statement (location: source ID 98, line 97, chars 3909-3932, hits: 0)`

It says this statement inside `isNative` isn't being covered, but it is
    function isNative(Currency currency) internal pure returns (bool) {
        return Currency.unwrap(currency) == Currency.unwrap(NATIVE);
    }

File: src/test/DynamicReturnFeeTestHook.sol [hunk 1]
```diff
@@ -24,6 +24,7 @@ contract DynamicReturnFeeTestHook is BaseTestHooks {
 
     function beforeSwap(address, PoolKey calldata, IPoolManager.SwapParams calldata, bytes calldata)
         external
+        view
```
Comment (hensha256) [inline src/test/DynamicReturnFeeTestHook.sol:27]: compiler warning on main rn

**Review by dianakocsis — CHANGES_REQUESTED** [review]

File: src/test/DynamicReturnFeeTestHook.sol (hunk 1 above)
Comment (dianakocsis) [reply src/test/DynamicReturnFeeTestHook.sol:27]: nice noticed this too

File: test/types/Currency.t.sol [hunk 2]
```diff
@@ -4,82 +4,133 @@ pragma solidity ^0.8.20;
...
+        assertEq(currencyTest.balanceOfSelf(erc20Currency), initialERC20Balance - amount);
+        assertEq(
+            currencyTest.balanceOfSelf(erc20Currency),
+            MockERC20(Currency.unwrap(erc20Currency)).balanceOf(address(currencyTest))
+        );
     }
 
-    function testCurrency_toId_native_returns0() public view {
-        assertEq(nativeCurrency.toId(), uint256(0));
+    function test_fuzz_balanceOf_native(uint256 amount) public {
```
Comment (dianakocsis) [inline test/types/Currency.t.sol:60]: can you assert that the currencyTest balance is the prev balance minus amount here too?

Comment (dianakocsis) [reply test/types/Currency.t.sol:60]: as well as `test_fuzz_balanceOf_token`, `test_fuzz_transfer_native`, and `test_fuzz_transfer_token`

**Review by dianakocsis — CHANGES_REQUESTED** [review]

**Review by dianakocsis — APPROVED** [review]

**Pull request merged**